capsules/extra/src/distance.rs

```diff
@@ -0,0 +1,195 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2024.
+
+//! Provides userspace with access to distance sensor.
+//!
+//! Userspace Interface
+//! -------------------
+//!
+//! ### `subscribe` System Call
+//!
+//! The `subscribe` system call supports the single `subscribe_number` zero,
+//! which is used to provide a callback that will return back the result of
+//! a distance sensor reading.
+//! The `subscribe` call return codes indicate the following:
+//!
+//! * `Ok(())`: the callback has been successfully been configured.
+//! * `ENOSUPPORT`: Invalid `subscribe_number`.
+//! * `NOMEM`: No sufficient memory available.
+//! * `INVAL`: Invalid address of the buffer or other error.
+//!
+//! ### `command` System Call
+//!
+//! The `command` system call supports one argument `cmd` which is used to
+//! specify the specific operation. Currently, the following commands are supported:
+//!
+//! * `0`: check whether the driver exists.
+//! * `1`: read the distance.
+//! * `2`: get the minimum distance that the sensor can measure based on the datasheet.
+//! * `3`: get the maximum distance that the sensor can measure based on the datasheet.
+//!
+//! The possible returns from the `command` system call indicate the following:
+//!
+//! * `Ok(())`: The operation has been successful.
+//! * `ENOSUPPORT`: Invalid `cmd`.
+//! * `NOMEM`: Insufficient memory available.
+//! * `INVAL`: Invalid address of the buffer or other error.
+//!
+//! Usage
+//! -----
+//!
+//! You need a device that provides the `hil::sensors::Distance` trait.
+//! Here is an example of how to set up a distance sensor with the HC-SR04.
```

Comment:
You really ought to make a component for the HC-SR04 distance sensor. https://github.com/tock/tock/tree/master/boards/components

I'm not sure what the best "example" would be here. Maybe the [HD44780 LCD](https://github.com/tock/tock/blob/master/boards/components/src/hd44780.rs) component, which includes GPIO pins and an Alarm, like you do.